src/mbed_tools/cli/configure.py

```diff
@@ -48,10 +48,11 @@ def configure(toolchain: str, mbed_target: str, program_path: str, mbed_os_path:
         program_path: the path to the local Mbed program
         mbed_os_path: the path to the local Mbed OS directory
     """
+    cmake_build_subdir = pathlib.Path(mbed_target.upper(), "develop", toolchain.upper())
```

Comment:
From the PR description and commit msg, it says that configs and builds should now live under `cmake_build/TARGET/PROFILE/TOOLCHAIN`, but this line makes `mbed-tools configure` always generate the `mbed_config.cmake` under `cmake_build/TARGET/develop/TOOLCHAIN`. Why is this?